maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultVersionRangeResolver.java

```diff
@@ -68,6 +70,15 @@
 @Named
 @Singleton
 public class DefaultVersionRangeResolver implements VersionRangeResolver, Service {
+    /**
+     * User property for version range handling. It may contain values of {@link Metadata.Nature} enum, or
+     * value "auto" to decide based on range: if any of the boundary version is snapshot, snapshots will be
+     * allowed, otherwise not. Default (unset) is "behave as before", query all repositories available in scope,
+     * so even snapshot repositories.
+     *
+     * @since 3.9.11
+     */
+    public static final String MAVEN_VERSION_RANGE_HANDLING = "maven.versionRangeResolutionNature";
```

Comment:
Usually, the constant name matches the value...

Reply:
Fixed, also improved javadoc